### DISCUSSION TITLE
Clearing stale readers

### DISCUSSION THREAD

**Discussion by pkaminski** [kriszyp/lmdb-js#284] [Q&A] [answered]
The LMDB docs warn against stale reader locks caused by process crashes (ones that don't cause a full restart and re-open the environment, that is).  The suggested mitigation is to call `mdb_reader_check` at regular intervals.  Is this needed and exposed in `lmdb-js`, or does the library take care of the problem automatically?

**Comment by kriszyp** [comment, marked as answer]
Yes, this is handled by the library automatically (on startup and periodically before write transactions).
